### DISCUSSION TITLE
requestSubscription is auto completing without prompting user to subscribe

### DISCUSSION THREAD

**Discussion by mj-schmitz** [hyochan/react-native-iap#2257] [Q&A]
I've had this problem for a few days now, where the requestSubscription function from the useIAP hook is not working properly but no error is thrown.  Essentially what's been happening is, only sometimes and somewhat randomly (usually the first 8-10 presses of subscribe button), the requestSubscription will run all the way through to the finally() block of the promise and not prompt the user to subscribe.

This makes it seem like our button is not responsive, but after 8-10 attempts it will load properly and prompt the user.  Here is the code:
```

import {
    useIAP,
    requestSubscription,
    IapIosSk2,
    setup,
} from 'react-native-iap';


const [subscriptionPending, setSubscriptionPending] = useState(false);
const [subscriptionError, setSubscriptionError] = useState(null);
const [isSubscriptionFlowActive, setIsSubscriptionFlowActive] = useState(false);

const handleBuySubscription = useCallback((productId, offerToken) => {
        if (isPlay && !offerToken) {
            console.warn(
                `There are no subscription Offers for selected product (Only required for Google Play purchases): ${productId}`
            );
        }
        console.log('handleBuySubscription - requesting subscription');
        setIsSubscriptionFlowActive(true);
        setSubscriptionPending(true);
        setSubscriptionError(null);
        requestSubscription({
            sku: productId,
            andDangerouslyFinishTransactionAutomaticallyIOS:true, //Only just recently added this, happens with out it as well.
            ...(offerToken && {
                subscriptionOffers: [{ sku: productId, offerToken }],

            })
        }).then((res) => {
            console.log(`handleBuySubscription - requestSubscription: ${JSON.stringify(res)}`); //hit's this then block 
        }).catch((error) => {
            console.log('handleBuySubscription - Error: ', error);
            setSubscriptionError(error);
        }).finally(() => {
            console.log('handleBuySubscription - requestSubscription Finished'); //finishes
            setIsSubscriptionFlowActive(false);
            setSubscriptionPending(false);
        });
    }, []);

```
When I add the andDangerouslyFinishTransactionAutomaticallyIOS field to it and this problem occurs, it returns an error that is 

```
undefined is not an object evaluation 't.id' 
```
But again, after 8 tries or so, the user is prompted for their subscription.  Very weird.

I have no idea why this bug is happening, and then only happening some of the time by not prompting the user immediately for their login info/asking them to subscribe.  Any ideas would be great, thank you.

**Comment by andresesfm** [comment]
I don't see any reference to `t.id` in the source code
